### PR TITLE
implement shape inference for concat

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
shape inference for concat

### CODE REVIEW

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 1]
```diff
@@ -1463,6 +1463,61 @@ bool ONNXConstantOp::inferShapes() {
...
+bool ONNXConcatOp::inferShapes() {
+  printf("hi alex\n");
+  int inputNum = getNumOperands();
+  for (int i = 0; i < inputNum; ++i) {
+    if (!getOperand(i).getType().cast<RankedTensorType>()) {
+      emitError("Input tensor(s) not ranked");
+      return false;
+    }
+  }
+  // Checking value of axis parameter/
```
Comment (doru1004) [inline src/Dialect/ONNX/ONNXOps.cpp:1478]: remove slash at end of comment.

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 2]
```diff
@@ -1463,6 +1463,61 @@ bool ONNXConstantOp::inferShapes() {
...
+  auto commonType = getOperand(0).getType().cast<RankedTensorType>();
+  auto commonShape = commonType.getShape();
+  auto commonRank = commonShape.size();
+  auto axisIndex = axis().getSExtValue();
+  if (!(axisIndex >= 0 && axisIndex < commonRank)) {
+    emitError("Concat axis value out of bound");
+    return false;
+  }
+  // Initial cummlative size is that of the first operand.
+  int cummulativeAxisSize = commonShape[axisIndex];
```
Comment (doru1004) [inline src/Dialect/ONNX/ONNXOps.cpp:1488]: Are we planning to handle the dynamic case at all?

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 3]
```diff
@@ -1463,6 +1463,61 @@ bool ONNXConstantOp::inferShapes() {
...
+  printf("hi alex\n");
+  int inputNum = getNumOperands();
+  for (int i = 0; i < inputNum; ++i) {
+    if (!getOperand(i).getType().cast<RankedTensorType>()) {
+      emitError("Input tensor(s) not ranked");
+      return false;
+    }
+  }
+  // Checking value of axis parameter/
+  auto commonType = getOperand(0).getType().cast<RankedTensorType>();
```
Comment (doru1004) [inline src/Dialect/ONNX/ONNXOps.cpp:1479]: Can we check that the element type is also the same across inputs?

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 4]
```diff
@@ -1463,6 +1463,61 @@ bool ONNXConstantOp::inferShapes() {
   return true;
 }
 
+//===----------------------------------------------------------------------===//
+// Concat
+
+bool ONNXConcatOp::inferShapes() {
+  printf("hi alex\n");
```
Comment (doru1004) [inline src/Dialect/ONNX/ONNXOps.cpp:1470]: Hello

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 1 above)
Comment (AlexandreEichenberger) [reply src/Dialect/ONNX/ONNXOps.cpp]: done

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 2 above)
Comment (AlexandreEichenberger) [reply src/Dialect/ONNX/ONNXOps.cpp:1487]: good point, restricted to static dimensions along the reduction for now

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 4 above)
Comment (AlexandreEichenberger) [reply src/Dialect/ONNX/ONNXOps.cpp]: so polite

**Pull request merged**